File: C++/2418. Sort the People.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
       map<int,string, greater<int>>mp;  // map to store in decending order   
    int n = names.size();
        for(int i=0; i<n; i++){
            mp[heights[i]] = names[i];
        }      
      vector<string>res;
        for(auto &p:mp){
            res.push_back(p.second);
        }
        return res;
    }
};

File: JAVA/2418. Sort the People.java
class Solution {
    public String[] sortPeople(String[] names, int[] heights) {
        int n = names.length;
        Map<Integer, String> mapping = new HashMap<>();
        for (int i = 0; i < n; ++i) {
            mapping.put(heights[i], names[i]);
        }
        Integer[] heightsArray = Arrays.stream(heights).boxed().toArray(Integer[]::new);
        Arrays.sort(heightsArray, (a, b) -> b - a);
        for (int i = 0; i < n; ++i) {
            names[i] = mapping.get(heightsArray[i]);
        }
        return names;
    }
}
